packages/fxa-settings/src/lib/model-data/data-stores/url-query-data.ts

```diff
@@ -20,11 +21,15 @@ export class UrlQueryData extends UrlData {
   protected setParams(params: URLSearchParams) {
     const url = new URL(this.window.location.href);
     url.search = params.toString();
-    // Use replaceState URL, but prevent page loads or history changes
-    this.window.history.replaceState({}, '', url.toString());
+
+    // Use replace false stops a page refresh
+    this.window.navigate(url.toString(), {
+      state: this.window.location.state,
```

Comment:
Should this be using a LocationStateData?

Reply:
We could do it this way, but I think it's better to have these two classes (UrlQueryData and LocationStateData) be independent. Creating a reference between them for this one action seems like overkill to me. This particular line doesn't even use the location.state, it just makes sure the current value persists. 

I think of each class as part of an access layer, so having them be indepedent seems good. Essentialy the classes both live at the same layer in the code that access window and or reach router window apis.